core/src/main/scala/org/apache/spark/rpc/netty/Dispatcher.scala

```diff
@@ -225,7 +225,15 @@ private[netty] class Dispatcher(nettyEnv: NettyRpcEnv, numUsableCores: Int) exte
           }
         }
       } catch {
-        case ie: InterruptedException => // exit
+        case _: InterruptedException => // exit
+        case t: Throwable =>
+          try {
+            // Re-submit a MessageLoop so that Dispatcher will still work if
+            // UncaughtExceptionHandler decides to not kill JVM.
+            threadpool.execute(new MessageLoop)
```

Comment:
should log before calling this?

Reply:
I usually don't log if it's going to be re-thrown since it will cause double logs.